bluesky/callbacks/zmq.py

```diff
@@ -43,8 +46,13 @@ def __init__(self, address, *, RE=None, zmq=None):
         self.hostname = socket.gethostname()
         self.pid = os.getpid()
         url = "tcp://%s:%d" % self.address
-        self._prefix = b'%s %d %d ' % (self.hostname.encode(),
-                                       self.pid, id(RE))
+        if prefix is not None:
```

Comment:
`prefix` will never be `None` because it was set to `''` above. I think the simplest approach is to make `prefix=''` the default in the `__init__` signature and then remove this if block.

Reply:
ah yes thanks! There is just one issue with the message format. It should probably be made homogeneous, have a delimiter. This is because the other end can't know what the format will be, and we probably don't want to mix different formats.
I'd vote for "foo\x00" (the null character)
In the docstring, we should mention the null character cannot be used in the prefix.

(and of course, all messages will have to come from matching versions so this will be incompatible with previous versions)